plugin/storage/es/spanstore/writer.go

```diff
@@ -151,6 +153,14 @@ func (s *SpanWriter) Close() error {
 
 func (s *SpanWriter) createIndex(indexName string, mapping string, jsonSpan *dbmodel.Span) error {
 	if !keyInCache(indexName, s.indexCache) {
+		s.indexMutex.Lock()
+		defer s.indexMutex.Unlock()
```

Comment:
This blocks not only index existence checks, but also all writes, which is not necessary.

Instead of mutex, we can use `sync.Once`, which is faster after the first execution.

Reply:
I don't think `sync.Once` would do, multiple indices are created from here, while `sync.Once` would only allow to create one. Do you have an example of how it would work?

Mutex is also only entered on cache miss, when index is missing.

Reply:
you're right, it would need to keep a map of these Once objects per index name, may not be worth it